src/plots/cartesian/axes.js

```diff
@@ -1070,7 +1187,93 @@ function roundDTick(roughDTick, base, roundingSet) {
 //      log with linear ticks: L# where # is the linear tick spacing
 //      log showing powers plus some intermediates:
 //          D1 shows all digits, D2 shows 2 and 5
-axes.autoTicks = function(ax, roughDTick) {
+axes.autoTicks = function(ax, roughDTick, isMinor) {
+    var majorDtick = ax._majorDtick;
+
+    if(isMinor && ax.type === 'date') {
+        if(typeof majorDtick === 'string' && majorDtick.charAt(0) === 'M') {
+            var months = Number(majorDtick.substring(1));
+
+            if(months > 12) {
+                ax.dtick = 'M12';
```

Comment:
This is too rigid - if the major tick is 50 years for example, we'll want minor at 10 years.

But more generally, I feel like the strategy here might be too hard-coded. I imagine the minor auto-tick process being the same as major except:
- using its own `nticks` and two neighboring major ticks as the "axis range"
- and then afterward check to see if the minor `dtick` you came up with is an even divisor of the major `dtick` - if not, adjust at that point.
  - Dates: if major is months and minor isn't, set minor to `'M1'`
  - Log: if major is a number or `'D?'` and minor is `'L?'`, set minor to `'D1'`
  - Both are numbers, `'M?'` or `'L?'` but `dtick / minor.dtick` isn't an integer: I think this will be pretty rare if major `dtick` was automatic, with perhaps a couple of exceptions:
    - 1-week or 2-week `dtick` and `minor.nticks < 7` -> set minor to 1 week
    - any other situation, maybe round `minor.nticks` down to 1, 2, 5, 10 etc and set `minor.dtick` to `dtick` divided by that rounded number?